Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -706,6 +724,26 @@ def get_current_speed(self, dir):
         with self.dllock:
             return self.curspeeds[dir]
 
+    def save_resume_data(self):
+        """
+        method to save resume data. It will return either :
+        stored resume data, or deferred if it's not available
+        """
```

Comment:
If you want less `instanceof(X, Deferred)` code you could let this function always return a deferred by using `defer.succeed` or use `maybeDeferred(save_resume_data)` on the caller.